### ISSUE TITLE
[BUG] <ZookeeperSyncDataService Method threw 'java.lang.StringIndexOutOfBoundsException' exception.>

### ISSUE THREAD

**Issue by wanghao123456** [apache/shenyu#4520]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When I click on the "synchronous sofa" button, the "unCacheSelectorData" and "unCacheRuleData" methods throw the following exception: "java.lang.StringIndexOutOfBoundsException".

![image](https://user-images.githubusercontent.com/23393153/228794663-921c0d4d-ec80-4457-97d8-9e1dd38fddd9.png)

### Steps To Reproduce

Reproducing the issue can be done by starting the "shenyu-examples-sofa" module and registering it to "shenyu", then clicking on the "synchronous sofa" button.

### Environment

```markdown
ShenYu version(s):2.6.0
zookeeper version(s):3.6.2
Update as of 2023-3-30 17:39:56.
```

### Debug logs

![image](https://user-images.githubusercontent.com/23393153/228798017-257c727a-f628-4e1d-8a1c-a82f0ec3dbdf.png)
![image](https://user-images.githubusercontent.com/23393153/228798528-fa4821c2-5a7b-4ee9-ba4b-dad1bb016d51.png)

2023-03-30 17:50:10 [Curator-TreeCache-4] ERROR org.apache.curator.framework.recipes.cache.TreeCache - 
java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.String.substring(String.java:1931)
	at org.apache.shenyu.sync.data.zookeeper.ZookeeperSyncDataService$MetadataCacheListener.event(ZookeeperSyncDataService.java:249)
	at org.apache.shenyu.sync.data.zookeeper.ZookeeperSyncDataService$AbstractDataSyncListener.childEvent(ZookeeperSyncDataService.java:180)
	at org.apache.curator.framework.recipes.cache.TreeCache$2.apply(TreeCache.java:760)
	at org.apache.curator.framework.recipes.cache.TreeCache$2.apply(TreeCache.java:754)
	at org.apache.curator.framework.listen.ListenerContainer$1.run(ListenerContainer.java:100)
	at org.apache.curator.shaded.com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
	at org.apache.curator.framework.listen.ListenerContainer.forEach(ListenerContainer.java:92)
	at org.apache.curator.framework.recipes.cache.TreeCache.callListeners(TreeCache.java:753)
	at org.apache.curator.framework.recipes.cache.TreeCache.access$1900(TreeCache.java:75)
	at org.apache.curator.framework.recipes.cache.TreeCache$4.run(TreeCache.java:865)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
2023-03-30 17:50:10 [Curator-TreeCache-2] ERROR org.apache.curator.framework.recipes.cache.TreeCache - 
java.lang.StringIndexOutOfBoundsException: String index out of range: -6
	at java.lang.String.substring(String.java:1967)
	at org.apache.shenyu.sync.data.zookeeper.ZookeeperSyncDataService.unCacheRuleData(ZookeeperSyncDataService.java:128)
	at org.apache.shenyu.sync.data.zookeeper.ZookeeperSyncDataService.access$800(ZookeeperSyncDataService.java:50)
	at org.apache.shenyu.sync.data.zookeeper.ZookeeperSyncDataService$RuleCacheListener.event(ZookeeperSyncDataService.java:296)
	at org.apache.shenyu.sync.data.zookeeper.ZookeeperSyncDataService$AbstractDataSyncListener.childEvent(ZookeeperSyncDataService.java:180)
	at org.apache.curator.framework.recipes.cache.TreeCache$2.apply(TreeCache.java:760)
	at org.apache.curator.framework.recipes.cache.TreeCache$2.apply(TreeCache.java:754)
	at org.apache.curator.framework.listen.ListenerContainer$1.run(ListenerContainer.java:100)
	at org.apache.curator.shaded.com.google.common.util.concurrent.DirectExecutor.execute(DirectExecutor.java:30)
	at org.apache.curator.framework.listen.ListenerContainer.forEach(ListenerContainer.java:92)
	at org.apache.curator.framework.recipes.cache.TreeCache.callListeners(TreeCache.java:753)
	at org.apache.curator.framework.recipes.cache.TreeCache.access$1900(TreeCache.java:75)
	at org.apache.curator.framework.recipes.cache.TreeCache$4.run(TreeCache.java:865)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

**Comment by yu199195** [comment]
other plugin have this bug？

**Comment by wanghao123456** [comment]
Yes, for example, "cache".

![image](https://user-images.githubusercontent.com/23393153/228809379-50496ca9-3b54-41ab-8c06-e2d70db8f4eb.png)

**Comment by moremind** [comment]
I will focus on this bug, please wait a minute

**Comment by moremind** [comment]
![image](https://user-images.githubusercontent.com/20184263/229293285-b1817bb3-13c3-4a8a-aacc-4ed1fee17b5a.png)
`shenyu/rule/sofa` is path, but shenyu can't analysis this path.

**Comment by moremind** [comment]
@wanghao123456 this bug has been fixed in #4525, pls use the new version! thanks for your response!
